utils/update_build_version.py

```diff
@@ -50,6 +54,23 @@ def command_output(cmd, dir):
     return stdout
 
 
+def deduceSoftwareVersion(dir):
+    """Returns a software version number parsed from the CHANGES file
+    in the given dir.
+
+    The CHANGES file describes most recent versions first.
+    """
+
+    pattern = re.compile('(v\d+\.\d+(wip)?) ')
```

Comment:
+1 to the suggestion on matching the date as well.
I was working from the conventions recommended for Perl modules:

https://metacpan.org/pod/release/BRICAS/CPAN-Changes-0.23/lib/CPAN/Changes/Spec.pod

For the preamble, it says "Any amount of text before the first Version line will be considered part of the preamble."
I should document that in the script.